code/modules/antagonists/demon/demons.dm

```diff
@@ -0,0 +1,172 @@
+	if(!L)
+		return
+	if(L.stat != DEAD) //demons however wont dust from the chapel so this needs to be a check to avoid spam while they're already dead
+		if(prob(50) && L.health >= 50)
+			switch(L.health)
+				if(85 to 100)
+					L.visible_message(span_warning("[L]'s skin begins to heat up and darken!"), span_danger("Your flesh begins to sear..."))
+				if(60 to 85)
+					L.visible_message(span_warning("[L]'s skin begins to melt apart!"), span_danger("Your skin is melting!"), "You hear sizzling.")
+			L.adjustFireLoss(5)
+		else if(L.health < 60)
+			if(!L.on_fire)
+				L.visible_message(span_warning("[L] lights up in a holy blaze!"), span_danger("Your skin catches fire!"))
+				L.emote("scream")
+			else
+				L.visible_message(span_warning("[L] continues to burn!"), span_danger("You continue to burn!"))
+			L.adjust_fire_stacks(5)
+			L.IgniteMob()
+	return
+
+/datum/antagonist/sinfuldemon/New()
+	. = ..()
+	demonsin = pick(demonsins)
+
+/datum/antagonist/sinfuldemon/proc/forge_objectives()
+	var/datum/objective/demonsingeneral/O
+	switch(demonsin)//the 5 most interesting of the 8 sins. Left out sloth because it sounds boring, couldn't think of a good enough objective/power for acedia, and lust for obvious reasons.
+		if(SIN_GLUTTONY)
+			O = new /datum/objective/demonsingeneral/gluttony
+		if(SIN_GREED)
+			O = new /datum/objective/demonsingeneral/greed
+		if(SIN_WRATH)
+			O = new /datum/objective/demonsingeneral/wrath
+			if(prob(50))
+				var/N = pick(/datum/objective/assassinate, /datum/objective/assassinate/cloned, /datum/objective/assassinate/once)
+				var/datum/objective/assassinate/kill_objective = new N
+				kill_objective.owner = owner
+				kill_objective.find_target()
+				objectives += kill_objective
+		if(SIN_ENVY)
+			O = new /datum/objective/demonsingeneral/envy
+			if(prob(50))
+				var/datum/objective/escape/escape_with_identity/identity_theft = new
+				identity_theft.owner = owner
+				identity_theft.find_target()
+				identity_theft.update_explanation_text()
+				objectives += identity_theft
+		if(SIN_PRIDE)
+			O = new /datum/objective/demonsingeneral/pride
+	objectives += O
+
+/datum/antagonist/sinfuldemon/can_be_owned(datum/mind/new_owner)
+	. = ..()
+	return . && (ishuman(new_owner.current) || iscyborg(new_owner.current))
+
+/proc/is_sinfuldemon(mob/living/M)
+	return M?.mind?.has_antag_datum(/datum/antagonist/sinfuldemon)
+
+/datum/antagonist/sinfuldemon/admin_add(datum/mind/new_owner,mob/admin)
+	var/choices = demonsins + "Random"
+	var/chosen_sin = input(admin,"What kind ?","Sin kind") as null|anything in choices
+	if(!chosen_sin)
+		return
+	if(chosen_sin in demonsins)
+		demonsin = chosen_sin
+	new_owner.add_antag_datum(src)
+	message_admins("[key_name_admin(admin)] has demonized [key_name_admin(new_owner)].")
+	log_admin("[key_name(admin)] has demonized [key_name(new_owner)].")
+
+/datum/antagonist/sinfuldemon/antag_listing_name()
+	return ..() + "(, demon of [demonsin])" // Boris Smith, demon of Wrath
```

Comment:
```suggestion
	return ..() + "(, demon of [demonsin])" // Boris Smith, demon of Wrath
```
Wouldn't this just end up being Boris Smith (, demon of Wrath)?
antag_listing_name only returns the realname